csharp/ql/src/Diagnostics/CompilerError.ql

```diff
@@ -0,0 +1,16 @@
+/**
+ * @name Compilation error
+ * @description A compilation error can cause extraction problems, and could lead to inaccurate results.
+ * @kind problem
+ * @problem.severity recommendation
+ * @precision medium
+ * @id cs/compilation-error
+ * @tags internal
+ */
+
+import csharp
+import semmle.code.csharp.commons.Diagnostics
+
+from CompilerError diagnostic
+select diagnostic,
+  diagnostic.getSeverityText() + " " + diagnostic.getTag() + " " + diagnostic.getFullMessage()
```

Comment:
I remember @aibaars mentioning that we should ideally use `$@` for the alert message due to storage concerns: not sure exactly what that would look like;
```
select diagnostic, "$@",
  diagnostic, diagnostic.getSeverityText() + " " + diagnostic.getTag() + " " + diagnostic.getFullMessage()
```
?

Reply:
I cannot see how that would help. Perhaps @aibaars could comment?

Reply:
It won't help. What could help is `"$@ $@ $@", "", diagnostic.getSeverityText(), "", diagnostic.getTag(), "", diagnostic.getFullMessage()`

The stored format of alert-resultsets makes use of string-pools, so the more strings are shared between rows the better. Like this all rows would share the message format string `"$@ $@ $@"`, and likely the severity and tag values come from a limited set which allows for a lot of sharing too.

Reply:
I don't really see why we should to this for this query and not other queries whose text is composed using `+`. Can we keep this code as it is?

Reply:
I wasn't thinking of this query in particular, but all the queries. I am happy with the alert text as-is, it is really for @aibaars to decide.

Reply:
@hvitved Feel free to merge this PR as-is. It might be worth revisiting all queries to use more format strings and arguments instead of string concatenation, but that can be done as a separate PR.